src/app/resize/resize.component.ts

```diff
@@ -0,0 +1,72 @@
+import {Component, ElementRef, EventEmitter, Input, Output} from "@angular/core";
+
+@Component({
+	selector: 'resize',
+	templateUrl: `
+		<div [class.overlay]="isExpanded"></div>
+		<div
+			class="handle"
+			[class.is-expanded]="isExpanded">
+		</div>
+		`,
+	styles: [`
+	.handle {
+		position: absolute;
+		right: 0;
+		top: 0;
+		height: 100%;
+		width: 15px;
+		border-right: 1px dashed rgba(128, 128, 128, 0.5);
+		cursor: col-resize;
+	}
+
+	.is-expanded {
+		width: 100%;
+	}
+
+	.overlay {
```

Comment:
this is pretty cool, but took me some time to understand, add a comment to explain the intent?